ucp/_libs/ucx_address.pyx

```diff
@@ -41,10 +63,7 @@ cdef class UCXAddress:
         cdef size_t length
         status = ucp_worker_get_address(ucp_worker, &address, &length)
         assert_ucs_status(status)
-        try:
-            return UCXAddress(int(<uintptr_t>address), length)
-        finally:
-            ucp_worker_release_address(ucp_worker, address)
+        return UCXAddress(int(<uintptr_t>address), length, worker=worker)
```

Comment:
It seems we are using `size_t` for `length` here, but the constructor takes `Py_ssize_t`. Should we consolidate to one of these two types?